From b5902e69c71bf556958f9fded99d1bcea2d7298c Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Wed, 7 Oct 2026 01:27:55 +0200
Subject: [PATCH] avformat/dash: Do not write into zero sized buffers in
 ff_dash_fill_tmpl_params()

The check is the one recommended in the report.

Fixes: lNi0cBMALYfT
Fixes: AISLE-2026-0111-00522
Fixes: AISLE-2026-0110-00522
Fixes: AISLE-2026-0109-00522
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
---
 libavformat/dash.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavformat/dash.c b/libavformat/dash.c
index fddfa82340a2..6c27fb578160 100644
--- a/libavformat/dash.c
+++ b/libavformat/dash.c
@@ -98,6 +98,8 @@ void ff_dash_fill_tmpl_params(char *dst, size_t buffer_size,
                                   int64_t time) {
     int dst_pos = 0;
     const char *t_cur = template;
+    if (!buffer_size)
+        return;
     while (dst_pos < buffer_size - 1 && *t_cur) {
         char format_tag[7]; // May be "%d", "%0Xd", or "%0Xlld" (for $Time$), where X is in [0-9]
         int n = 0;
